src/ophyd_async/epics/_backend/common.py

```diff
@@ -10,9 +12,15 @@ def get_supported_values(
     if not datatype:
         return {x: x or "_" for x in pv_choices}
 
-    if not issubclass(datatype, str):
+    if issubclass(datatype, RuntimeEnum):
+        if not datatype.choices.issubset(frozenset(pv_choices)):
+            raise TypeError(
+                f"{pv} has choices {pv_choices}, "
+                f"which do not match RuntimeEnum, which has {datatype.choices}"
+            )
+    elif not issubclass(datatype, str):
         raise TypeError(f"{pv} is type Enum but doesn't inherit from String")
-    if issubclass(datatype, Enum):
+    elif issubclass(datatype, Enum):
```

Comment:
The original was correct with the `if`, but a little confusing.

I suggest:
```python
    if issubclass(datatype, Enum):
        # Do an exhaustive check
        if not issubclass(datatype, str):
            raise TypeError(f"{pv} is type Enum but doesn't inherit from String")
        choices = tuple(v.value for v in datatype)
        if set(choices) != set(pv_choices):
            raise TypeError(
                (
                    f"{pv} has choices {pv_choices}, "
                    f"which do not match {datatype}, which has {choices}"
                )
            )
        return {x: datatype(x) for x in pv_choices}
    elif issubclass(datatype, RuntimeEnum):
        # Do a subset check
        if not datatype.choices.issubset(frozenset(pv_choices)):
            raise TypeError(
                f"{pv} has choices {pv_choices}, "
                f"which do not match RuntimeEnum, which has {datatype.choices}"
            )
    return {x: x for x in pv_choices}
```

Reply:
Don't we want to check it's a string even if it's not an Enum? This change would allow:

```
        get_supported_values("", MyEnum, ("test",))
```

Reply:
I don't understand. We should support 3 cases:
- `class MyEnum(str, Enum)`: exhaustive check, `signal.get_value()` returns `MyEnum`
- `MyRuntimeEnum = RuntimeEnum["one", "two"]` (or `SubsetEnum` if we rename it): subset check, `signal.get_value()` returns str
- None: no check, `signal.get_value()` returns str

Reply:
Hmmm I'm inclined to think we should have it fail outright here if it's not a `RuntimeEnum` or `Enum`.